### PR TITLE
Hook unittests

### PR DESCRIPTION
**Pull request by ronykoz**
## Status
In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/15782

## Description
Added tests for the validation hook, and refactored them a bit.
Still need to add test for the main flow and the diff files, and will add docstrings for everything after first approval.
Also removed id_set.json and moved it's creation to Circle.

Please note that after the merge of this PR **all** content developers will to add the path to their content repo to PYTHONPATH, and update their hooks accordingly.
Everyone will also have to restart their IDE so that the hooks that are running from the IDE will be able to load the new PYTHONPATH

### CODE REVIEW

File: Tests/secrets_white_list.json [hunk 1]
```diff
@@ -662,6 +662,8 @@
   },
   "generic_strings":
   [
+    "self.conf_json_validator",
+    "self.IMAGE_MAX_SIZE:",
```
Comment (yardensade) [inline Tests/secrets_white_list.json]: Currently works on in lowercase, either lower case it or add .lower() when comparing string against the whitelist.

File: Tests/scripts/hook_validations/script.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+       yaml_data (dict): the data of the script in dictionary format.
+    """
+    def __init__(self, file_path, check_git=True):
+        self._is_valid = True
+        self.file_path = file_path
+
+        if check_git:
+            self.change_string = run_git_command("git diff HEAD {0}".format(self.file_path))
+            with open(file_path, 'r') as file_data:
+                self.yaml_data = yaml.safe_load(file_data)
```
Comment (yaakovi) [inline Tests/scripts/hook_validations/script.py]: if `check_git` is False, self.yaml_data will not be defined for all class methods

File: Tests/scripts/hook_validations/script.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                existing_args.append(arg_name)
+
+            else:
+                self._is_valid = False
+                return True
+
+        return False
+
+    def is_arg_changed(self):
+        """Check if the argument as been changed."""
```
Comment (yaakovi) [inline Tests/scripts/hook_validations/script.py]:
```suggestion
        """Check if the argument has been changed."""
```

File: Tests/scripts/hook_validations/integration.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        """Check if a command name or argument as been changed.
+
+        Returns:
+            bool. Whether a command name or argument as been changed.
+        """
+        current_command_to_args = self._get_command_to_args(self.current_integration)
+        old_command_to_args = self._get_command_to_args(self.old_integration)
+
+        for command, args_dict in old_command_to_args.items():
+            if command not in current_command_to_args.keys() or \
```
[the remark this answers is not in the document]
Comment (ronykoz) [reply Tests/scripts/hook_validations/integration.py]: We still need to make sure the command is there although it is deprecated, no?

**Comment by ronykoz** [comment]
Completed tests:
- [ ] Changed integration package
- [ ] Changed command name or arg
- [ ] Changed integration context
- [ ] Changed integration docker image
- [ ] Added required field
- [ ] All of the above for scripts
- [ ] Renaming of integration

File: Tests/scripts/hook_validations/image.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import glob
+
+from Tests.test_utils import *
+from Tests.scripts.constants import IMAGE_REGEX, INTEGRATION_REGEX
+
+
+class ImageValidator(object):
```
[the remark this answers is not in the document]
Comment (yaakovi) [reply Tests/scripts/hook_validations/image.py]: i had it as an image, my bad...

File: Tests/scripts/hook_validations/integration.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (yaakovi) [reply Tests/scripts/hook_validations/integration.py]: @glicht @anara123
will we ever remove deprecated commands from integrations or will they stay as deprecated forever? if so, there is nothing to be fixed.

File: .circleci/config.yml [hunk 6]
```diff
@@ -2,7 +2,7 @@ version: 2
 jobs:
   build:
     docker:
-      - image: devdemisto/content-build:1.0.0.132 # disable-secrets-detection
+      - image: devdemisto/content-build:1.0.0.159 # disable-secrets-detection
```
Comment (yardensade) [inline .circleci/config.yml]: You could take off the disable secrets comment as we now skip docker images :)

**Pull request merged**